coderd/workspaceapps.go

```diff
@@ -122,8 +123,7 @@ func (api *API) workspaceApplicationAuth(rw http.ResponseWriter, r *http.Request
 		return
 	}
 
-	// Encrypt the API key.
-	encryptedAPIKey, err := api.AppSecurityKey.EncryptAPIKey(workspaceapps.EncryptedAPIKeyPayload{
+	encryptedAPIKey, err := jwtutils.Encrypt(ctx, api.AppEncryptionKeyCache, workspaceapps.EncryptedAPIKeyPayload{
```

Comment:
You're missing the 1 minute expiration.

While we're at it, we should set a NotBefore to 1 minute in the past (to allow for minor clock skew between coderd and the proxy), an issuer (coderd) and an audience (wsproxy), and check all of these on decryption.

Reply:
🤦 good catch